File: src/main/java/org/taktik/connector/technical/validator/EhealthReplyValidator.java
package org.taktik.connector.technical.validator;

import org.taktik.connector.technical.exception.SoaErrorException;
import be.fgov.ehealth.commons.protocol.v1.ResponseType;
import be.fgov.ehealth.commons.protocol.v2.StatusResponseType;

public interface EhealthReplyValidator {
   boolean validateReplyStatus(ResponseType var1) throws SoaErrorException;

   boolean validateReplyStatus(StatusResponseType var1) throws SoaErrorException;
}

File: src/main/kotlin/org/taktik/connector/technical/validator/EhealthReplyValidator.kt
package org.taktik.connector.technical.validator

import org.taktik.connector.technical.exception.SoaErrorException
import be.fgov.ehealth.commons.protocol.v1.ResponseType
import be.fgov.ehealth.commons.protocol.v2.StatusResponseType

interface EhealthReplyValidator {
    @Throws(SoaErrorException::class)
    fun validateReplyStatus(response: ResponseType): Boolean

    @Throws(SoaErrorException::class)
    fun validateReplyStatus(response: be.fgov.ehealth.commons._1_0.protocol.ResponseType): Boolean

    @Throws(SoaErrorException::class)
    fun validateReplyStatus(response: StatusResponseType): Boolean
}
